src/www/model/server.ts

```diff
@@ -38,10 +38,30 @@ export interface Server {
   checkReachable(): Promise<boolean>;
 }
 
+// Represents a Shadowsocks proxy server configuration.
+export interface ShadowsocksConfig {
+  host?: string;
+  port?: number;
+  password?: string;
+  method?: string;
+  name?: string;
+}
+
+// Endpoint to retrieve proxy configurations.
+export interface ProxyConfigSource {
+  url: string;
+}
+
+// Server configuration.
+export interface ServerConfig {
+  proxy?: ShadowsocksConfig;
+  source?: ProxyConfigSource;
+  name?: string;
+}
+
```

Comment:
Should we move the proxy config fetching logic from `OutlineServer` to `App` by defining:

```typescript
interface ProxyConfigResponse {
  proxies?: ShadowsocksConfig[];
  redirectUrl?: string;
  httpStatusCode: number;
}

interface Server {
  fetchProxyConfig(): Promise<ProxyConfigResponse>
}
```